### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by bernabasy**
### In this pull request, we have made the following changes
Using Devise for user authentication
The following changes are made in this branch:

- Generated views and models with device
- Created registration parameters in ApplicationController
- Implemented authentication features
- Modified views and layouts

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (julie-ify) [inline app/models/user.rb]: - [ ] Kindly add `confirmable` to the devise configurations as we need to confirm a user after sign up.

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <%= f.password_field :password, autocomplete: "new-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
```
Comment (julie-ify) [inline app/views/devise/registrations/new.html.erb:29]: - [x] I was unable to sign up as a user due to the posts counter error below. Kindly fix that.

  image

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<h2>Log in</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <div class="field">
+  <%= f.label :name %><br />
+  <%= f.text_field :name %>
+</div>
+
```
Comment (julie-ify) [inline app/views/devise/sessions/new.html.erb]: - [x] User should be able to login with only email and password. Kindly remove this name field

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @bernabasy and @PeteUgwu,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/models/user.rb]: Fixed.

File: app/views/devise/sessions/new.html.erb (hunk 3 above)
Comment (Abenezer-Tilahun) [reply app/views/devise/sessions/new.html.erb]: Fixed.

File: app/views/devise/registrations/new.html.erb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply app/views/devise/registrations/new.html.erb:29]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @bernabasy @PeteUgwu 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git -Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bernabasy** [comment]
Thanks bro

**Pull request merged**